Marlin/src/module/endstops.cpp

```diff
@@ -615,7 +615,8 @@ void Endstops::update() {
   #define UPDATE_ENDSTOP_BIT(AXIS, MINMAX) SET_BIT_TO(live_state, _ENDSTOP(AXIS, MINMAX), (READ(_ENDSTOP_PIN(AXIS, MINMAX)) != _ENDSTOP_INVERTING(AXIS, MINMAX)))
   #define COPY_LIVE_STATE(SRC_BIT, DST_BIT) SET_BIT_TO(live_state, DST_BIT, TEST(live_state, SRC_BIT))
 
-  #if BOTH(G38_PROBE_TARGET, HAS_Z_MIN_PROBE_PIN) && NONE(CORE_IS_XY, CORE_IS_XZ, MARKFORGED_XY)
+  #if ENABLED(G38_PROBE_TARGET) && EITHER(HAS_Z_MIN_PROBE_PIN, USES_Z_MIN_PROBE_PIN) && NONE(CORE_IS_XY, CORE_IS_XZ, MARKFORGED_XY)
+    #define HAS_G38_PROBE 1
```

Comment:
Yeah, so here we have `HAS_Z_MIN_PROBE_PIN`, which requires `HAS_BED_PROBE`, so the `G38` probe has got to be the same probe as the bed probe. (_i.e._, There is only ever just one probe.)

Meanwhile, `USES_Z_MIN_PROBE_PIN` is simply a reflection of the user's pin preference from the configuration, and it doesn't really guarantee anything, except that if we got here the sanity-check must have passed.

I get the sense from this PR that something "hackish" is going on, which only works because you are making a core change that allows the `Z_MIN_PROBE_PIN` to be defined even when the probe is actually using the `Z_MIN_PIN`. I'm not sure whether it is a good idea to yank on this thread.